### PR TITLE
Export direct_reclaims_total metric to prometheus

### LINKED ISSUE #136: Export "direct_reclaims"
There is a memcached bug where this error appears:
```
level=debug ts=2022-06-23T10:28:27.220696365Z caller=memcached_client.go:406 name=frontend-cache msg="failed to store item to memcached" key=1@b2ae91c4319dafc4 sizeBytes=86848 server=10.70.1.208:11211 err="memcache: unexpected response line from \"set\": \"SERVER_ERROR out of memory storing object\\r\\n\""
```

The above is logged from [here](https://github.com/thanos-io/thanos/blob/a0f41812f01e782dd9e5e09755f0348da6fd8e88/pkg/cacheutil/memcached_client.go#L406-L413).

Background (from [related issue](https://github.com/grafana/mimir/issues/2235)):

Memcached 1.5 and above uses a segmented LRU by default ([blog post](https://memcached.org/blog/modern-lru/)). Items can be evicted by a background routine if they’re expired, or directly during a mset operation if the cache memory is full. The latter operation is called a “direct reclaim”.

Query-frontend caches results with 7d TTL. Since the load test has run for less than 7d, presumably all evictions are caused by direct reclaims. Using the [stat command](https://github.com/memcached/memcached/blob/master/doc/protocol.txt) on memcached shows:
0 reclaimed (Number of times an entry was stored using memory from an expired entry)
0 crawler_reclaimed (Total items freed by LRU Crawler)
983890 direct_reclaims (Times worker threads had to directly reclaim or evict items)

We would like the above “direct_reclaims” metric to be exported in Prometheus.

### PR DESCRIPTION
**Pull request by Vandit1604**
fixes #136

### CODE REVIEW

File: pkg/exporter/exporter.go [hunk 1]
```diff
@@ -277,6 +278,12 @@ func New(server string, timeout time.Duration, logger log.Logger, tlsConfig *tls
 			nil,
 			nil,
 		),
+		directReclaims: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, "", "total_direct_reclaims"),
```
Comment (SuperQ) [inline pkg/exporter/exporter.go:282]: Must match Prometheus naming conventions.

```suggestion
			prometheus.BuildFQName(Namespace, "", "direct_reclaims_total"),
```

**Review by SuperQ — CHANGES_REQUESTED** [review]

File: pkg/exporter/exporter.go (hunk 1 above)
Comment (Vandit1604) [reply pkg/exporter/exporter.go]: Done

**Comment by Vandit1604** [comment]
@SuperQ Is there anything else that needs to be done here to complete the PR?

**Review by Sticksman — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -45,6 +45,8 @@ The exporter collects a number of statistics from the server:
 # TYPE memcached_current_connections gauge
 # HELP memcached_current_items Current number of items stored by this instance.
 # TYPE memcached_current_items gauge
+# HELP memcached_total_direct_reclaims Times worker threads had to directly reclaim or evict items.
+# TYPE memcached_total_direct_reclaims counter
```
Comment (SuperQ) [inline README.md:49]: This also needs to be updated.

Comment (Vandit1604) [reply README.md]: Done.

**Review by matthiasr — APPROVED** [review]

**Pull request merged**